agent/consul/leader_connect_ca.go

```diff
@@ -1139,81 +1136,113 @@ func (c *CAManager) runRenewIntermediate(ctx context.Context) error {
 // RenewIntermediate checks the intermediate cert for
 // expiration. If more than half the time a cert is valid has passed,
 // it will try to renew it.
-func (c *CAManager) RenewIntermediate(ctx context.Context, isPrimary bool) error {
+func (c *CAManager) RenewIntermediate(ctx context.Context) error {
+	_, err := c.renewIntermediate(ctx, false, false)
+	return err
+}
+
+func (c *CAManager) renewIntermediateNow(ctx context.Context) error {
+	renewed, err := c.renewIntermediate(ctx, true, true)
+	if err != nil {
+		return err
+	}
+	if !renewed {
+		return errors.New("expected to force renew and did not")
+	}
+	return nil
+}
+
+func (c *CAManager) renewIntermediate(
+	ctx context.Context,
+	forceRenewAheadOfSchedule bool,
+	skipAsyncWatch bool,
+) (bool, error) {
 	// Grab the 'lock' right away so the provider/config can't be changed out while we check
 	// the intermediate.
 	if _, err := c.setState(caStateRenewIntermediate, true); err != nil {
-		return err
+		return false, err
 	}
 	defer c.setState(caStateInitialized, false)
 
+	isPrimary := c.serverConf.InPrimaryDatacenter()
+
 	provider, _ := c.getCAProvider()
 	if provider == nil {
 		// this happens when leadership is being revoked and this go routine will be stopped
-		return nil
+		return false, nil
 	}
 	// If this isn't the primary, make sure the CA has been initialized.
 	if !isPrimary && !c.secondaryHasProviderRoots() {
-		return fmt.Errorf("secondary CA is not yet configured.")
+		return false, fmt.Errorf("secondary CA is not yet configured.")
 	}
 
 	state := c.delegate.State()
 	_, root, err := state.CARootActive(nil)
 	if err != nil {
-		return err
+		return false, err
 	}
 	activeRoot := root.Clone()
 
 	// If this is the primary, check if this is a provider that uses an intermediate cert. If
 	// it isn't, we don't need to check for a renewal.
 	if isPrimary && !primaryUsesIntermediate(provider) {
-		return nil
+		return false, nil
 	}
 
 	activeIntermediate, err := provider.ActiveIntermediate()
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	if activeIntermediate == "" {
-		return fmt.Errorf("datacenter doesn't have an active intermediate.")
+		return false, fmt.Errorf("datacenter doesn't have an active intermediate.")
 	}
 
 	intermediateCert, err := connect.ParseCert(activeIntermediate)
 	if err != nil {
-		return fmt.Errorf("error parsing active intermediate cert: %v", err)
+		return false, fmt.Errorf("error parsing active intermediate cert: %v", err)
 	}
 
-	if lessThanHalfTimePassed(c.timeNow(), intermediateCert.NotBefore, intermediateCert.NotAfter) {
-		return nil
+	if !forceRenewAheadOfSchedule {
+		if lessThanHalfTimePassed(c.timeNow(), intermediateCert.NotBefore, intermediateCert.NotAfter) {
+			return false, nil
```

Comment:
Is this the only time we return `(false, nil)`? Wondering if we can alternatively return a new err